package/src/package.rs

```diff
@@ -0,0 +1,296 @@
+    /// executing the program.
+    pub dependencies: Vec<Dependency>,
+}
+
+/// A procedure exported by a package, along with its digest and
+/// signature(will be added after MASM type attributes are implemented).
+#[derive(Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct PackageExport {
+    /// The fully-qualified name of the procedure exported by this package
+    pub name: QualifiedProcedureName,
+    /// The digest of the procedure exported by this package
+    #[cfg_attr(test, proptest(value = "Digest::default()"))]
+    pub digest: Digest,
+}
+
+impl fmt::Debug for PackageExport {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let Self { name, digest } = self;
+        f.debug_struct("PackageExport")
+            .field("name", &format_args!("{}", name))
+            .field("digest", &format_args!("{}", DisplayHex::new(&digest.as_bytes())))
+            .finish()
+    }
+}
+
+// PACKAGE
+// ================================================================================================
+
+/// A package containing a [Program]/[Library], and a manifest (exports and dependencies).
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Package {
+    /// Name of the package
+    pub name: String,
+    /// The MAST artifact ([Program] or [Library]) of the package
+    pub mast: MastArtifact,
+    /// The package manifest, containing the set of exported procedures and their signatures,
+    /// if known.
+    pub manifest: PackageManifest,
+}
+
+impl Package {
+    // const MAGIC: &'static [u8] = b"MASP\0";
+    // const FORMAT_VERSION: &'static [u8] = b"1.0\0";
+
+    ///// Parses a package from the provided bytes
+    //pub fn read_from_bytes<B>(bytes: B) -> Result<Self, Report>
+    //where
+    //    B: AsRef<[u8]>,
+    //{
+    //    use alloc::borrow::Cow;
+    //
+    //    let bytes = bytes.as_ref();
+    //
+    //    let bytes = bytes
+    //        .strip_prefix(Self::MAGIC)
+    //        .ok_or_else(|| Report::msg("invalid package: missing header"))?;
+    //    let bytes = bytes.strip_prefix(Self::FORMAT_VERSION).ok_or_else(|| {
+    //        Report::msg(format!(
+    //            "invalid package: incorrect version, expected '1.0', got '{}'",
+    //            bytes.get(0..4).map(String::from_utf8_lossy).unwrap_or(Cow::Borrowed("")),
+    //        ))
+    //    })?;
+    //
+    //    bitcode::deserialize(bytes).map_err(Report::msg)
+    //}
+    //
+    ///// Serializes the package into a byte array
+    //pub fn write_to_bytes(&self) -> Result<Vec<u8>, Report> {
+    //    let mut bytes = Vec::new();
+    //    bytes.extend_from_slice(Self::MAGIC);
+    //    bytes.extend_from_slice(Self::FORMAT_VERSION);
+    //    let mut data = bitcode::serialize(self).map_err(Report::msg)?;
+    //    bytes.append(&mut data);
+    //    Ok(bytes)
+    //}
```

Comment:
Is this meant to be commented-out?

Reply:
Yes, I missed it. Thanks for spotting this! 
In my defense, I planned to give this PR a "polish" before marking it ready and asking for another round of reviews. 🙂

Reply:
Done.